### PR TITLE
Read resource settings from environment variables

### PR DESCRIPTION
**Pull request by cjs**
Settings such as `OTEL_RESOURCE_ATTRIBUTES` were not being read in from environment variables. Having this available is important for some automated environments where `OTEL_RESOURCE_ATTRIBUTES` is auto-generated from deployment tooling.

Note that setting `service.name` in `OTEL_RESOURCE_ATTRIBUTES` or via `OTEL_SERVICE_NAME` will still be a no-op because there is a default value set from the command line flags. 

This also bumps the https://github.com/open-telemetry/opentelemetry-go libraries to the latest stable release. This was required to get it to work with our LightStep environment, although I'm not entirely sure what in the changelog was breaking. 

Finally, the https://pkg.go.dev/go.opentelemetry.io/otel/sdk@v1.6.3/resource#WithTelemetrySDK detector was added to inject version information about the OpenTelemetry SDK libraries into span attributes. This is to aid administration of observability services.

### CODE REVIEW

File: otel.go [hunk 1]
```diff
@@ -124,7 +124,7 @@ func OpenTelemetryRunE(flagPrefix string, prerunLevel zerolog.Level) CobraRunFun
 }
 
 func initOtelTracer(exporter trace.SpanExporter, serviceName string, propagators []string) error {
-	res, _ := resource.New(context.Background(), resource.WithAttributes(semconv.ServiceNameKey.String(serviceName)))
+	res, _ := setResource(serviceName)
```
Comment (jzelinskie) [inline otel.go]: This function does return an error. Any reason why we shouldn't propagate it?

File: otel_test.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+				string(semconv.ServiceNameKey):           "testservice",
+				string(semconv.DeploymentEnvironmentKey): "test",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			for k, v := range tt.envVars {
+				t.Setenv(k, v)
+			}
```
Comment (jzelinskie) [inline otel_test.go:53]: We probably should defer calls to `os.Unsetenv()` to cleanup for the next test.

**Review by jzelinskie — CHANGES_REQUESTED** [review]

File: otel_test.go (hunk 2 above)
Comment (cjs) [reply otel_test.go:53]: https://pkg.go.dev/testing#T.Setenv says that it does an automatic cleanup, but we can't use parallel.

Comment (jzelinskie) [reply otel_test.go:53]: Oh! I didn't notice this was calling it on `t.`! my bad!

File: otel.go (hunk 1 above)
Comment (cjs) [reply otel.go]: resolved in b9a185a67b564e54370c4de13ef7d73bd2c827a7

File: otel.go [hunk 3]
```diff
@@ -124,7 +124,11 @@ func OpenTelemetryRunE(flagPrefix string, prerunLevel zerolog.Level) CobraRunFun
 }
 
 func initOtelTracer(exporter trace.SpanExporter, serviceName string, propagators []string) error {
-	res, _ := resource.New(context.Background(), resource.WithAttributes(semconv.ServiceNameKey.String(serviceName)))
+	res, err := setResource(serviceName)
+
```
Comment (jzelinskie) [inline otel.go]: tiny nitpick: drop this new line

**Review by jzelinskie — APPROVED** [review]

**Pull request merged**